pkg/controller/ps/status.go

```diff
@@ -269,13 +271,24 @@ func appHost(ctx context.Context, cl client.Reader, cr *apiv1alpha1.PerconaServe
 	return host, nil
 }
 
-func appStatus(ctx context.Context, cl client.Reader, size int32, labels map[string]string, version string) (apiv1alpha1.StatefulAppStatus, error) {
+func (r *PerconaServerMySQLReconciler) appStatus(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL, compName string, size int32, labels map[string]string, version string) (apiv1alpha1.StatefulAppStatus, error) {
 	status := apiv1alpha1.StatefulAppStatus{
 		Size:  size,
 		State: apiv1alpha1.StateInitializing,
 	}
 
-	pods, err := k8s.PodsByLabels(ctx, cl, labels)
+	sfsObj := &appsv1.StatefulSet{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: cr.Name + "-" + compName, Namespace: cr.Namespace}, sfsObj)
```

Comment:
we have `<comp>.Name` function, it's better to use it here

Reply:
I know, but that means that here we would have to have a bunch of ifs or a switch to check what comp it is. 
I want to have something like `<comp>.Status` that will handle the status for a particular component. That would be much more clear, but I left it out just for now because of time, since there's a bit more to refactor and I really want to merge this one already.

I'll refactor it in my subsequent PR.

Reply:
the caller of the function already passes the comp name, we just need to call proper `Name()` function and pass its value to this function

Reply:
so rather than `r.appStatus(ctx, cr, "mysql", ...)` we'll call `r.appStatus(ctx, cr, mysql.Name(cr), ...)`

Reply:
`<comp>.Name` returns `cr.Name + "-" + componentName` and we need only the `componentName`.